lib/Conversion/TTIRToD2M/TTIRToD2M.cpp

```diff
@@ -360,13 +385,105 @@ class D2MNamedElementwiseRewriter final
 
     auto [origInputs, origOutputs] =
         splitDpsSignature(adaptor, op.getDpsInits().size());
+
+    // Check if this op supports scalar operands (only add, mul, sub, div, pow)
+    constexpr bool supportsScalar =
+        std::is_same_v<ConcreteOp, ttir::AddOp> ||
+        std::is_same_v<ConcreteOp, ttir::MultiplyOp> ||
+        std::is_same_v<ConcreteOp, ttir::SubtractOp> ||
+        std::is_same_v<ConcreteOp, ttir::DivOp> ||
+        std::is_same_v<ConcreteOp, ttir::PowOp>;
+
+    // Check for scalar operands (rank 0 tensors) by checking converted operand
+    // types
+    std::optional<FloatAttr> scalarAttr = std::nullopt;
+    SmallVector<Value> tensorInputs;
+    SmallVector<Operation *> scalarOpsToErase;
+
+    for (Value origInput : origInputs) {
+      // Check the converted operand's type for rank (from adaptor)
+      auto convertedTensorType =
+          mlir::dyn_cast<mlir::RankedTensorType>(origInput.getType());
+
+      // Check if this is a scalar tensor (rank 0)
+      if (convertedTensorType && convertedTensorType.getRank() == 0) {
+        if constexpr (!supportsScalar) {
+          return rewriter.notifyMatchFailure(
+              op, "Scalar operands only supported for add, multiply, subtract, "
+                  "div, and pow ops");
+        }
+
+        // Extract the scalar constant value from the converted operand
+        auto *convertedDefiningOp = origInput.getDefiningOp();
+
+        // Handle d2m.full (already converted)
+        if (auto d2mFullOp =
+                mlir::dyn_cast_or_null<d2m::FullOp>(convertedDefiningOp)) {
+          auto fillValueAttr = d2mFullOp.getFillValue();
+          if (auto floatAttr = mlir::dyn_cast<FloatAttr>(fillValueAttr)) {
+            scalarAttr = floatAttr;
+            // Track the d2m.full op for erasure
+            scalarOpsToErase.push_back(d2mFullOp);
+            continue; // Skip adding this to tensorInputs
+          } else if (auto intAttr =
```

Comment:
:warning: **llvm\-else\-after\-return** :warning:
do not use `` else `` after `` continue ``
```suggestion
          } if (auto intAttr =
```